### PR TITLE
Make tx messages/actions typesafe and wireup front page

### LINKED ISSUE #954: Make tx messages/actions typesafe and wireup front page

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #954

### CODE REVIEW

**Comment by sorawit** [comment]
@Benzbeeb Please fix the conflict. I moved end block logic to its own file here https://github.com/bandprotocol/bandchain/commit/fe1499a2586ff109080f324ead83e0fa9aaf781c. Fixing should be simple but make sure you understand what's going on. Let me know if you have question.

**Review by sorawit — CHANGES_REQUESTED** [review]
Fix conflict

File: scan/src/subscriptions/TxSub.re [hunk 1]
```diff
@@ -1,3 +1,391 @@
...
+
+  let getDescription = msg => {
+    switch (msg) {
+    | Send(send) =>
+      (send.amount |> Coin.toCoinsString) ++ "->" ++ (send.toAddress |> Address.toBech32)
+    | CreateDataSource(dataSource) => dataSource.name
+    | EditDataSource(dataSource) => dataSource.name
+    | CreateOracleScript(oracleScript) => oracleScript.name
+    | EditOracleScript(oracleScript) => oracleScript.name
+    | Request(_) => "yo"
```
Comment (sorawit) [inline scan/src/subscriptions/TxSub.re]: What is this?

File: scan/src/subscriptions/TxSub.re [hunk 2]
```diff
@@ -1,3 +1,391 @@
...
+    //   | None => ""
+    //   }
+    // | None => "?"
+    // }
+    | Report(report) => "yo"
+    //   switch (msg.events->Event.getValueOfKey("report.code_name")) {
+    //   | Some(value) => "#" ++ (report.requestID |> string_of_int) ++ " " ++ value
+    //   | None => "?"
+    //   }
+    | AddOracleAddress(_) => "ADDORACLEADDRESS DESCRIPTION"
```
Comment (sorawit) [inline scan/src/subscriptions/TxSub.re]: Also here. Why is it still placeholder?

File: scan/src/subscriptions/TxSub.re [hunk 3]
```diff
@@ -1,3 +1,391 @@
...
+      | "edit_validator" => EditValidator(json |> EditValidator.decode)
+      | _ => Unknown
+      }
+    );
+  };
+  let decodeActions = json => json |> JsonUtils.Decode.list(decodeAction);
+
+  let getRoute = msg =>
+    switch (msg) {
+    | Send(_) => None
```
Comment (sorawit) [inline scan/src/subscriptions/TxSub.re]: Also here. Shouldn't this ticket wire-up everything?

File: scan/src/subscriptions/TxSub.re [hunk 4]
```diff
@@ -66,6 +454,35 @@ module TxCountConfig = [%graphql
 |}
 ];
 
+module Event = {
```
Comment (sorawit) [inline scan/src/subscriptions/TxSub.re]: Why is it here? Where is this getting used?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/subscriptions/TxSub.re [hunk 5]
```diff
@@ -90,6 +507,7 @@ let getList = (~page, ~pageSize, ()) => {
       MultiConfig.definition,
       ~variables=MultiConfig.makeVariables(~limit=pageSize, ~offset, ()),
     );
+
```
Comment (taobun) [inline scan/src/subscriptions/TxSub.re]: Why need empty line here?

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]
LGTM

**Review by taobun — APPROVED** [review]

**Pull request merged**